eclipse-che/che-plugin-registry
# fix(entrypoint): Fix image regexp when using CHE_SIDECAR_CONTAINERS_REGISTRY_URL

### What does this PR do?
1. Fix problems with image regexp:
   - single quotes not supported
   - multiline not supported
   - multiple occurence
2. Adding unit tests script

### Screenshot/screencast of this PR
N/A

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/19405 and https://github.com/eclipse/che/issues/19492

### How to test this PR?
1. Build the registry from this branch:
   ```
   ./build.sh
   ```
3. run the built container and check that  meta.yaml generated are OK, for instance `/var/www/html/v3/plugins/ms-python/python/latest/meta.yaml`
   ```
   podman run -t quay.io/eclipse/che-plugin-registry:nightly cat /var/www/html/v3/plugins/ms-python/python/latest/meta.yaml
   ```
   should contain these lines
    ```yaml
      - image: 'quay.io/eclipse/che-plugin-sidecar@sha256:64311210e710ca1614839f897285fc1d39b2ca17215d56c165d091aafd19a14b'
   ```
4. check that meta.yaml works well with the env variable `CHE_SIDECAR_CONTAINERS_REGISTRY_URL`
   ```
   podman run -e CHE_SIDECAR_CONTAINERS_REGISTRY_URL=https://fakeregistry.io:5000 -t quay.io/eclipse/che-plugin-registry:nightly cat /var/www/html/v3/plugins/ms-python/python/latest/meta.yaml
   ```
   should contain these lines
    ```yaml
    - image: 'https://fakeregistry.io:5000/eclipse/che-plugin-sidecar@sha256:64311210e710ca1614839f897285fc1d39b2ca17215d56c165d091aafd19a14b'
   ```

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.
